### PR TITLE
Tags defined during health check registration are now available in th…

### PR DESCRIPTION
**Pull request by viktorpeacock**
…e health report entries.

Summary of the changes (Less than 80 chars)
 - HealthReportEntry has tags defined in the HealthCheck registration. Updated unit test.

Addresses #9378.

@rynowak , please let me know if more unit tests are needed.

### CODE REVIEW

**Comment by viktorpeacock** [comment]
I've never had to do a pull request into aspnet repo. Can somebody please advise on what I can do about the following error? 

> Generated code is not up to date. You might need to regenerate the reference assemblies or project list (see docs/ReferenceAssemblies.md and docs/ReferenceResolution.md)

I read the readme files and not sure what extra dependency I've introduced. System namespace was already referenced in the HealthReport class. Am I missing something?

**Comment by rynowak** [comment]
@viktorpeacock - your wish is my command:

You may need to add dotnet to your path first, you can do this by running:

`. .\activate.ps1`

From the root of the repo. Note the leading dot. 

https://github.com/aspnet/Extensions/blob/master/docs/ReferenceAssemblies.md

If you try this and you're still stuck I can just do it for you 😆

File: src/HealthChecks/Abstractions/src/HealthReportEntry.cs [hunk 1]
```diff
@@ -22,13 +23,15 @@ public struct HealthReportEntry
         /// <param name="duration">A value indicating the health execution duration.</param>
         /// <param name="exception">An <see cref="Exception"/> representing the exception that was thrown when checking for status (if any).</param>
         /// <param name="data">Additional key-value pairs describing the health of the component.</param>
-        public HealthReportEntry(HealthStatus status, string description, TimeSpan duration, Exception exception, IReadOnlyDictionary<string, object> data)
+        /// <param name="tags">Tags associated with the health check that generated the report entry.</param>
+        public HealthReportEntry(HealthStatus status, string description, TimeSpan duration, Exception exception, IReadOnlyDictionary<string, object> data, IEnumerable<string> tags = null)
```
Comment (rynowak) [inline src/HealthChecks/Abstractions/src/HealthReportEntry.cs]: We can't change an existing constructor like this as it's a binary breaking change.

Add a new constructor with the extra parameter and update the old one to call the new one.

File: src/HealthChecks/HealthChecks/test/DefaultHealthCheckServiceTest.cs [hunk 2]
```diff
@@ -57,6 +57,9 @@ public async Task CheckAsync_RunsAllChecksAndAggregatesResultsAsync()
             const string UnhealthyMessage = "Halp!";
             const string HealthyMessage = "Everything is A-OK";
             var exception = new Exception("Things are pretty bad!");
+            IEnumerable<string> healthyCheckTags = new List<string> { "healthy-check-tag" };
+            IEnumerable<string> degradedCheckTags = new List<string> { "degraded-check-tag" };
+            IEnumerable<string> unhealthyCheckTags = new List<string> { "unhealthy-check-tag" };
```
Comment (rynowak) [inline src/HealthChecks/HealthChecks/test/DefaultHealthCheckServiceTest.cs]: use `var`

**Comment by viktorpeacock** [comment]
@rynowak I made the changes. The code check is failing again - I don't understand why. I followed instructions in the readme and ran dotnet msbuild /t:GenerateReferenceSource. This resulted in no changes.

Running .\eng\scripts\GenerateReferenceAssemblies.ps1 results in many modified files across multiple projects.

**Comment by rynowak** [comment]
Sorry this is causing problems. Sent you a PR: https://github.com/viktorpeacock/Extensions/pull/1

**Comment by viktorpeacock** [comment]
@rynowak please let me know if any further changes are needed. 

Thank you for all your help!

**Comment by rynowak** [comment]
Thanks!

**Pull request merged**